js/extensions/bindings/datatableBinding.js

```diff
@@ -209,8 +209,24 @@ define([
 			// drawing may access observables, which updating we do not want to trigger a redraw to the table
 			// see: https://knockoutjs.com/documentation/computed-dependency-tracking.html#IgnoringDependencies
 			ko.ignoreDependencies(table.draw);
+
+			jQuery.fn.dataTableExt.search.pop();
```

Comment:
The table shouldn't know anything about its contents. Seems like very specific and tightly-coupled solution. I would look into `HTML5 data-* attributes` or mechanism which stays behind it:
https://datatables.net/examples/advanced_init/html5-data-attributes.html